MARC.js

```diff
@@ -541,7 +542,35 @@ record.prototype.translate = function(item) {
 		this._associateNotes(item, "545", "ab");
 		
 		// Extract title
-		this._associateDBField(item, "245", "abn", "title");
+		//  a = main title
+		//  b = subtitle
+		//  n = Number of part/section of a work
+		//  p = Name of part/section of a work
+		var titlesubfields = this.getFieldSubfields("245")[0];
+		item.title = titlesubfields["a"];
+		if (titlesubfields["b"]) {
+			//If the title did not end in a punctuation mark, we should add a colon
+			if (item.title.search(/[?:,.!;]\s*$/) == -1) {
+				item.title += ':';
+			}
+			item.title += ' ' + titlesubfields["b"];
+		}
+		if (titlesubfields["n"]) {
+			//If the title did not end in a punctuation mark, we should add a point
+			if (item.title.search(/[?:,.!;]\s*$/) == -1) {
+				item.title += '.';
+			}
+			item.title += ' ' + titlesubfields["n"];
+		}
+		if (titlesubfields["p"]) {
+			//If the title did not end in a punctuation mark, we should add a point
+			if (item.title.search(/[?:,.!;]\s*$/) == -1) {
+				item.title += ':';
```

Comment:
what about if we only have $p and not $n -- that's not implausible, is it? In that case, two colons right after each other would seem odd, no?